lib/ace/virtual_renderer.js

```diff
@@ -838,8 +838,10 @@ var VirtualRenderer = function(container, theme) {
             this.$updateScrollBarV();
             if (changes & this.CHANGE_H_SCROLL)
                 this.$updateScrollBarH();
-            this.$gutterLayer.element.style.marginTop = (-config.offset) + "px";
-            this.content.style.marginTop = (-config.offset) + "px";
+            
+            dom.translate(this.$gutterLayer.element, 0, -config.offset);
```

Comment:
On windows transforms break antialiasing, and on my mac laptop screen i don't see significant difference in performance, compared to adding `contain:strict` on layer elements.
Is different more significant on larger screens?

Reply:
In theory transform will bypass paint and go straight into compositing. Can you try https://stackoverflow.com/questions/6492027/css-transform-jagged-edges-in-chrome and see if that fixes it?

Reply:
It indeed bypasses painting, and that breaks subpixel antialiasing. Setting `backface-visibility:hidden;` or using other hacks does not seem to help.

Setting `will-change: transform` or `contain: strict` changes behavior of marginTop/top to skip paint as well, but sadly it has similar effect on antialiasing.

On windows the positive effect of this is very small anyway, because we do not have smooth scrolling, and when scroll amount is large enough to change lines we trigger full layout and repaint by removing the line